### PR TITLE
[ONNX] Add quantization support to _avg_pool opset 9 and clean up

### PR DESCRIPTION
**Pull request by justinchuby**
- Add quantization support to _avg_pool opset 9
- Clean up reused / unused variables in avgpool helper
- Add types
- Sort `__all__`

### CODE REVIEW

File: torch/onnx/symbolic_opset10.py [hunk 1]
```diff
@@ -145,12 +146,12 @@ def _avg_pool(name, tuple_fn):
...
     def symbolic_fn(
         g,
-        input,
-        kernel_size,
-        stride,
-        padding,
-        ceil_mode,
-        count_include_pad,
+        input: _C.Value,
+        kernel_size: Tuple[int, ...],
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset10.py:150]: parsed by "is", so `List` would be more accurate

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -56,270 +56,270 @@
...
+    "upsample_trilinear3d",
+    "var_mean",
+    "var",
+    "view_as",
+    "view",
+    "where",
+    "wrap_logical_op_with_cast_to",
+    "wrap_logical_op_with_negation",
+    "zeros_like",
+    "zeros",
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:322]: what's going on with these changes?

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -1412,17 +1411,20 @@ def symbolic_fn(
                 mode_s="constant",
                 value_f=0.0,
             )
-            padding = (0,) * len(padding)
+            result_padding = (0,) * len(padding)
         if ceil_mode:
-            padding = padding + tuple(a + b for (a, b) in zip(padding_ceil, padding))
+            padding_ceil = get_pool_ceil_padding(input, kernel_size, stride, padding)
+            result_padding = padding + tuple(
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:1417]: just by looking at the code, there is potential behavior change. If in old code `padding` was modified in above `count_include_pad` branch, it is now captured in `result_padding`.

File: torch/onnx/symbolic_helper.py [hunk 4]
```diff
@@ -1129,13 +1129,17 @@ def _batchnorm_helper(g, input, weight, bias, running_mean, running_var):
...
 
-def _avgpool_helper(tuple_fn, padding, kernel_size, stride, divisor_override, name):
+def _avgpool_helper(
+    tuple_fn: Callable[[Any], Sequence[int]],
+    padding: Union[int, Tuple[int, ...]],
+    kernel_size,
+    stride,
+    divisor_override,
+    name,
+) -> Tuple[int]:
```
Comment (BowenBao) [inline torch/onnx/symbolic_helper.py:1139]: `-> Tuple[int, ...]`

File: torch/onnx/symbolic_helper.py [hunk 5]
```diff
@@ -1129,13 +1129,17 @@ def _batchnorm_helper(g, input, weight, bias, running_mean, running_var):
     return weight, bias, running_mean, running_var
 
 
-def _avgpool_helper(tuple_fn, padding, kernel_size, stride, divisor_override, name):
+def _avgpool_helper(
+    tuple_fn: Callable[[Any], Sequence[int]],
```
Comment (BowenBao) [inline torch/onnx/symbolic_helper.py:1133]: nit: would it be better to be consistent regarding `Sequence` vs `Tuple`?

File: torch/onnx/symbolic_opset10.py [hunk 6]
```diff
@@ -145,12 +146,12 @@ def _avg_pool(name, tuple_fn):
...
+        stride: Tuple[int, ...],
+        padding: Union[int, Tuple[int, ...]],
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset10.py:152]: is it possible for padding to have type `int`? It's also parsed by `is`.

**Review by BowenBao — COMMENTED** [review]
Consider mention the added quantization support in title as well.

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (justinchuby) [reply torch/onnx/symbolic_opset9.py:322]: They are sorted

File: torch/onnx/symbolic_helper.py (hunk 5 above)
Comment (justinchuby) [reply torch/onnx/symbolic_helper.py:1133]: Changed to Sequence

File: torch/onnx/symbolic_opset10.py (hunk 6 above)
Comment (justinchuby) [reply torch/onnx/symbolic_opset10.py]: Changed to Sequence[int]

File: torch/onnx/symbolic_opset10.py (hunk 1 above)
Comment (justinchuby) [reply torch/onnx/symbolic_opset10.py]: Changed to Sequence which can accomodate the List output from parse_args

File: torch/onnx/symbolic_opset9.py (hunk 3 above)
Comment (justinchuby) [reply torch/onnx/symbolic_opset9.py:1424]: Good catch will update

Comment (justinchuby) [reply torch/onnx/symbolic_opset9.py:1424]: Updated

**Comment by justinchuby** [comment]
> Consider mention the added quantization support in title as well.

Done. Thanks!

**Review by BowenBao — APPROVED** [review]
🚀

**Pull request closed without merging**